### ISSUE TITLE
Incorrect general_settings_to_kit migration in elementor v3.x.x

### ISSUE THREAD

**Issue by bigdigital** [elementor/elementor#12298]
Hi!
The migration function _v_3_0_0_move_general_settings_to_kit has a bug in $settings_to_slider convertation.
Normally it should convert settings value to an array with [unit, size] but this migration does not work if the value in settings was set to 0. This happens because migration  function has a condition 
`if ( ! empty( $current_settings[ $setting ] ) ) (line 690) .`
 So if the value equal to 0, the migrations would'not be executed.
To replicate this, install the elementor v2.x.x, set the space_between_widgets to 0 and update elementor to v3.x.x. 
In result, the kit META_KEY would be like this (expected array in **space_between_widgets** field  )

```
array (
  'default_generic_fonts' => 'Sans-serif',
  'container_width' =>
  array (
    'unit' => 'px',
    'size' => '1200',
  ),
  **'space_between_widgets' => '0',**
  'global_image_lightbox' => 'yes',
  'viewport_md' => '700',
  'viewport_lg' => '900',
  'custom_colors' =>
  array (
    0 =>
    array (
      '_id' => 'ed4f6bc',
      'title' => 'Saved Color #1',
      'color' => '#000',
    ),
    1 =>
    array (
      '_id' => '6cd4fa5c',
      'title' => 'Saved Color #2',
      'color' => '#FFF',
    ),
    2 =>
    array (
      '_id' => '56f34bd',
      'title' => 'Saved Color #3',
      'color' => '#283875',
    ),
    3 =>
    array (
      '_id' => '3bec5641',
      'title' => 'Saved Color #4',
      'color' => '#666666',
    ),
    4 =>
    array (
      '_id' => '59a6c3a4',
      'title' => 'Saved Color #5',
      'color' => '#AAAAAA',
    ),
    5 =>
    array (
      '_id' => '40e95124',
      'title' => 'Saved Color #6',
      'color' => '#000000',
    ),
    6 =>
    array (
      '_id' => '438994a4',
      'title' => 'Saved Color #7',
      'color' => 'RGBA(255, 255, 255, 0.6)',
    ),
  ),
)
```

Due to this bug, the settings change to its default values, in this case, the  space between widgets settings would be changed from 0 to 20px

Regards,
Artem

**Comment by VadymPogorelov** [comment]
Would it be much more logical to set the "space_between_widgets" parameter to 0 by default? After all, this is a page builder, not a text editor. Why are there extra gaps between blocks in the Builder? I think most users immediately set this parameter to 0

**Comment by MichaelDarko** [comment]
Can confirm that widget gap set to 0 turns into 20px after update.

**Comment by shilo-ey** [comment]
This issue has been resolved in Elementor v3.0.5

Feel free to update

Thanks!
